src/main/java/com/fasterxml/jackson/databind/cfg/DefaultCacheProvider.java

```diff
@@ -10,10 +24,100 @@ public class DefaultCacheProvider
 {
     private static final long serialVersionUID = 1L;
 
-    public static DefaultCacheProvider defaultInstance() {
-        return new DefaultCacheProvider();
+    /**
+     * {@link LookupCache} instance used to create {@link DeserializerCache} instance when {@link #forDeserializerCache} is invoked.
+     */
+    protected final LookupCache<JavaType, JsonDeserializer<Object>> _deserializerCache;
```

Comment:
No. Provider should not pre-construct this; an instance should be created when requested, based on configured size.